### DISCUSSION TITLE
Automatische Integration aller verfügbaren Interfaces

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#335] [General]
Im Issue #328 und speziell im [Kommentar](https://github.com/danielperna84/hahomematic/issues/328#issuecomment-1070746712)
ist davon die Rede, bei der Einrichtung der Integration diese automatisch erkennen zu lassen welche Interfaces verfügbar sind (BidCos-RF, HmIP-RF usw.) und diese Interfaces entsprechend automatisch einzurichten / zu nutzen.

Grundsätzlich gefällt mir das, macht es doch die Einrichtung noch simpler / komfortabler speziell für Nutzer die mit den technischen Hintergründen (verschiedene Protokolle für verschiedene Homematic Generationen) nichts am Hut haben.
Außerdem würden mögliche Fehlkonfigurationen wie eventuell hier (#329) von vornherein ausgeschlossen. 

Aber:
Nicht jeder möchte möglicherweise "alles was geht" auch nutzen.
Wer z.B. nur 5 Heizgruppen hat will vielleicht nur diese in HA sehen weil ihn die Einzelgeräte nicht so interessieren.

Was ich mir vorstellen könnte, sollte diese "Auto-Integration" kommen, wäre vielleicht eine Checkbox die per default aktiv (Auto Integration) ist. Will man das nicht nutzen deaktiviert man die Option und kann dann wie bisher die gewünschten Interfaces manuell anwählen.

**Comment by danielperna84** [comment]
Ich glaube der Erkennung verfügbarer Interfaces ging es um was anderes. Grundsätzlich kann es ziemlich exotische Setups geben, bei denen man immer manuell konfigurieren muss. Ich weiß von einem User, der eine zweite CCU durch ein VPN angebunden hatte, und deshalb dann auch Ports nutzen musste die nicht default sind, und eben auch mit der Callback-IP arbeiten. Und die Möglichkeit sowas in HA einzubinden sollte nicht verloren gehen. Wenn hier also noch etwas voran getrieben wird um das Setup zu erleichtern, dann (hoffentlich) nicht auf Kosten der Flexibilität.
